frame/contracts/src/benchmarking/mod.rs

```diff
@@ -1583,6 +1587,786 @@ benchmarks! {
 		), vec![], Endow::Max)?;
 		let origin = RawOrigin::Signed(instance.caller.clone());
 	}: call(origin, instance.addr, 0u32.into(), Weight::max_value(), vec![])
+
+	// We make the assumption that pushing a constant and dropping a value takes roughly
+	// the same amount of time. We follow that `t.load` and `drop` both have the weight
+	// of this benchmark / 2. We need to make this assumption because there is no way
+	// to measure them on their own using a valid wasm module. We need their individual
+	// values to derive the weight of individual instructions (by substraction) from
+	// benchmarks that include those for parameter pushing and return type dropping.
+	// We call the weight of `t.load` and `drop`: `w_param`.
+	// The weight that would result from the respective benchmark we call: `w_bench`.
+	//
+	// w_i{32,64}const = w_drop = w_bench / 2
+	instr_i64const {
+		let r in 0 .. INSTR_BENCHMARK_BATCHES;
+		use body::DynInstr::{RandomI64Repeated, Regular};
+		let mut sbox = Sandbox::from(&WasmModule::<T>::from(ModuleDefinition {
+			call_body: Some(body::repeated_dyn(r * INSTR_BENCHMARK_BATCH_SIZE, vec![
+				RandomI64Repeated(1),
+				Regular(Instruction::Drop),
+			])),
+			.. Default::default()
+		}));
+	}: {
+		sbox.invoke();
+	}
+
+	// w_i{32,64}load = w_bench - 2 * w_param
+	instr_i64load {
+		let r in 0 .. INSTR_BENCHMARK_BATCHES;
+		use body::DynInstr::{RandomUnaligned, Regular};
+		let mut sbox = Sandbox::from(&WasmModule::<T>::from(ModuleDefinition {
+			memory: Some(ImportedMemory::max::<T>()),
+			call_body: Some(body::repeated_dyn(r * INSTR_BENCHMARK_BATCH_SIZE, vec![
+				RandomUnaligned(0, code::max_pages::<T>() * 64 * 1024 - 8),
+				Regular(Instruction::I64Load(3, 0)),
+				Regular(Instruction::Drop),
+			])),
+			.. Default::default()
+		}));
+	}: {
+		sbox.invoke();
+	}
+
+	// w_i{32,64}store{...} = w_bench - 2 * w_param
+	instr_i64store {
+		let r in 0 .. INSTR_BENCHMARK_BATCHES;
+		use body::DynInstr::{RandomI64Repeated, RandomUnaligned, Regular};
```

Comment:
could just use `body::DynInstr::*` somewhere above

Reply:
Makes sense. It is used a lot.